agent/acl_test.go

```diff
@@ -188,14 +185,9 @@ func TestACL_Version8(t *testing.T) {
 	t.Parallel()
 
 	t.Run("version 8 disabled", func(t *testing.T) {
-		resolveFn := func(string) (structs.ACLIdentity, acl.Authorizer, error) {
-			require.Fail(t, "should not have called delegate.ResolveToken")
-			return nil, nil, fmt.Errorf("should not have called delegate.ResolveToken")
-		}
-
 		a := NewTestACLAgent(t, t.Name(), TestACLConfig()+`
  		acl_enforce_version_8 = false
- 	`, resolveFn)
+ 	`, nil, nil)
```

Comment:
I changed things to make not specifying the callback have a default implementation which returns only errors saying it wasn't expected to be used.